src/test/java/uk/gov/pay/webhooks/webhook/resource/WebhookResourceIT.java

```diff
@@ -206,6 +216,111 @@ public void notFoundShouldReturn404() {
                 .statusCode(Response.Status.NOT_FOUND.getStatusCode());
     }
 
+    @Test
+    public void shouldDeleteSomeWebhookMessages() {
+        var webhookExternalId = "a-webhook-external-id";
+        WebhookMessageExternalIds webhookMessageExternalIds = setupWebhookWithMessagesExpectedToBePartiallyDeleted(webhookExternalId);
+        List<String> expectedWebhookExternalIdsNotDeleted = webhookMessageExternalIds.notDeleted;
+        List<String> expectedWebhookMessageExternalIds = setupThreeWebhookMessagesThatShouldNotBeDeleted(); // maxAgeOfMessages=7 so these webhook messages should not be deleted
+
+        given().port(port)
+                .contentType(JSON)
+                .post("/v1/webhook/tasks/delete_messages")
+                .then()
+                .statusCode(200);
+
+        WebhookMessageSearchResponse webhookMessageSearchResponse = given().port(port)
+                .get(format("/v1/webhook/%s/message", webhookExternalId))
+                .then()
+                .body("results.size()", is(expectedWebhookExternalIdsNotDeleted.size() + expectedWebhookMessageExternalIds.size()))
+                .extract().body().as(WebhookMessageSearchResponse.class);
+        
+        Collection<String> expectedWebhookExternalIds = 
+                CollectionUtils.union(expectedWebhookMessageExternalIds, expectedWebhookExternalIdsNotDeleted);
+        
+        assertThat(expectedWebhookExternalIds,
+                everyItem(in(webhookMessageSearchResponse.results().stream().map(WebhookMessageResponse::externalId).toList())));
+
+        expectedWebhookExternalIds.forEach(webhookMessageExternalId ->
+                given().port(port)
+                        .get(format("/v1/webhook/%s/message/%s/attempt", webhookExternalId, webhookMessageExternalId))
+                        .then()
+                        .body("size()", is(1)));
+        
+        webhookMessageExternalIds.deleted.forEach(webhookMessageExternalId ->
+                given().port(port)
+                        .get(format("/v1/webhook/%s/message/%s", webhookExternalId, webhookMessageExternalId))
+                        .then()
+                        .statusCode(Response.Status.NOT_FOUND.getStatusCode()));
+    }
+
+    private List<String> setupThreeWebhookMessagesThatShouldNotBeDeleted() {
+        DateFormat df = new SimpleDateFormat("yyyy-MM-dd");
+        String date = df.format(Date.from(OffsetDateTime.now().minusDays(1).toInstant()));
+        List<String> webhookMessageExternalIds = List.of("thirteenth-message-external-id", "fourteenth-message-external-id", "fifteenth-message-external-id");
+        app.getJdbi().withHandle(h -> h.execute("""
+                            INSERT INTO webhook_messages VALUES
```

Comment:
Can this SQL be in DbTestHelper?

Reply:
I can't see any existing DbTestHelper class so do you mean a new class?
I'm following the [existing pattern](https://github.com/alphagov/pay-webhooks/blob/main/src/test/java/uk/gov/pay/webhooks/webhook/resource/WebhookResourceIT.java#L221-L260) here.

Reply:
It is named `DatabaseTestHelper`.